combat-trainer.lic

```diff
@@ -1313,19 +1316,22 @@ class TrainerProcess
     DRRoom.dead_npcs.size > dead_count
   end
 
-  def analyze(game_state)
+  def analyze(game_state,fail_count)
     return if game_state.npcs.empty?
+    return if fail_count > @analyze_retry_count
     return if DRSkill.getxp('Tactics') >= 32
     case bput('analyze', 'roundtime', 'Analyze what', 'Your analysis reveals a massive opening', /You reveal an? .* [weakness|opening]/, /Your analysis reveals an? .* [weakness|opening]/, 'You fail to find any holes', 'There is nothing else', 'Face what', 'You must be closer', 'You must be standing', 'Strangely, you don\'t feel like fighting right now', 'flying too high for you to attack') # unless
     when 'Analyze what'
       fput('face next')
-      analyze(game_state)
+      analyze(game_state,fail_count)
     when 'You must be closer'
       fput('engage')
     when 'Your analysis reveals a massive opening'
       return
-    when /You reveal an? .* [weakness|opening]/, /Your analysis reveals an? .* [weakness|opening]/, 'You fail to find any holes'
-      analyze(game_state)
+    when 'You fail to find any holes'
+      analyze(game_state,fail_count+1)
+    when /You reveal an? .* [weakness|opening]/, /Your analysis reveals an? .* [weakness|opening]/
+      analyze(game_state,fail_count)
```

Comment:
Good question.  I personally don't have a preference, since increasing the value of the setting would have a pretty similar practical effect.